src/ui/MainMenu.cpp

```diff
@@ -3985,25 +4166,1324 @@ namespace MainMenu {
+		                        auto prompt = main_menu_frame->findFrame("mono_prompt");
+		                        if (prompt) {
+			                        auto dimmer = static_cast<Frame*>(prompt->getParent()); assert(dimmer);
+			                        dimmer->removeSelf();
+		                        }
+		                        beginFade(FadeDestination::RootMainMenu);
+                            }
+                        );
+					    multiplayer = SINGLE;
+					    disconnectFromLobby();
+					    pauseGame(2, 0);
+				    } else {
+				        //TODO announce player disconnect
+					    //char shortname[32] = { 0 };
+					    //strncpy(shortname, stats[net_packet->data[16]]->name, 22);
+					    //newString(&lobbyChatboxMessages, 0xFFFFFFFF, language[1376], shortname);
+			            createInviteButton(playerDisconnected);
+				    }
+				    continue;
+			    }
+
+			    // got a chat message
+			    else if (packetId == 'CMSG') {
+				    //newString(&lobbyChatboxMessages, 0xFFFFFFFF, (char*)(&net_packet->data[4]));
+				    // TODO print lobby message
+				    playSound(238, 64);
+				    continue;
+			    }
+
+			    // update svFlags
+			    else if (packetId == 'SVFL') {
+				    lobbyWindowSvFlags = SDLNet_Read32(&net_packet->data[4]);
+				    continue;
+			    }
+
+			    // keepalive
+			    else if (packetId == 'KPAL') {
+				    continue; // just a keep alive
+			    }
+
+                // error
+			    else {
+			        printlog("Got a mystery packet: %c%c%c%c",
+			            (char)net_packet->data[0],
+			            (char)net_packet->data[1],
+			            (char)net_packet->data[2],
+			            (char)net_packet->data[3]);
+			        continue;
+			    }
+			}
+		}
+	}
+
+	static void handleNetwork() {
+	    if (multiplayer == SERVER) {
+	        handlePacketsAsServer();
+	    } else if (multiplayer == CLIENT) {
+	        handlePacketsAsClient();
+	    }
+        doKeepAlive();
+	}
+
+	static void setupNetGameAsServer() {
+	    // allocate data for client connections
+	    net_clients = (IPaddress*) malloc(sizeof(IPaddress) * MAXPLAYERS);
+	    net_tcpclients = (TCPsocket*) malloc(sizeof(TCPsocket) * MAXPLAYERS);
+	    for (int c = 0; c < MAXPLAYERS; c++) {
+		    net_tcpclients[c] = NULL;
+	    }
+
+	    // allocate packet data
+	    net_packet = SDLNet_AllocPacket(NET_PACKET_SIZE);
+	    assert(net_packet);
+
+        // setup game
+	    clientnum = 0;
+	    multiplayer = SERVER;
+	    if (loadingsavegame) {
+		    loadGame(clientnum);
+	    }
+	}
+
+	static void finalizeOnlineLobby() {
+	    closeNetworkInterfaces();
+	    directConnect = false;
+	    if (LobbyHandler.getHostingType() == LobbyHandler_t::LobbyServiceType::LOBBY_STEAM) {
+#ifdef STEAMWORKS
+		    for ( int c = 0; c < MAXPLAYERS; c++ ) {
+			    if ( steamIDRemote[c] ) {
+				    cpp_Free_CSteamID(steamIDRemote[c]);
+				    steamIDRemote[c] = NULL;
+			    }
+		    }
+		    ::currentLobbyType = k_ELobbyTypePrivate;
```

Comment:
Invalid assignment here - two different enums. I'm sure you want to CreateLobby() using k_ELobbyTypePrivate but ::currentLobbyType should be something like `LobbyType::LobbyOnline`?

Reply:
::currentLobbyType refers to the definition in steam.hpp, so this compiles fine for me

Reply:
My b, still was using code from Friday testing out steamworks builds now